lib/membrane_http_adaptive_stream/hls.ex

```diff
@@ -153,25 +169,89 @@ defmodule Membrane.HTTPAdaptiveStream.HLS do
     end
   end
 
-  defp serialize_track(%Manifest.Track{} = track) do
+  defp serialize_track(%Track{} = track) do
     target_duration = Ratio.ceil(track.target_segment_duration / Time.second()) |> trunc()
+    supports_ll_hls? = Track.supports_partial_segments?(track)
+
+    target_partial_duration =
+      if supports_ll_hls? do
+        Float.ceil(Ratio.to_float(track.target_partial_segment_duration / Time.second()), 3)
+      else
+        nil
+      end
 
     """
     #EXTM3U
     #EXT-X-VERSION:#{@version}
     #EXT-X-TARGETDURATION:#{target_duration}
-    #EXT-X-MEDIA-SEQUENCE:#{track.current_seq_num}
-    #EXT-X-DISCONTINUITY-SEQUENCE:#{track.current_discontinuity_seq_num}
-    #EXT-X-MAP:URI="#{track.header_name}"
-    #{track.segments |> Enum.flat_map(&serialize_segment/1) |> Enum.join("\n")}
-    #{if track.finished?, do: "#EXT-X-ENDLIST", else: ""}
-    """
+    """ <>
+      serialize_ll_hls_tags(supports_ll_hls?, target_partial_duration) <>
+      """
+      #EXT-X-MEDIA-SEQUENCE:#{track.current_seq_num}
+      #EXT-X-DISCONTINUITY-SEQUENCE:#{track.current_discontinuity_seq_num}
+      #EXT-X-MAP:URI="#{track.header_name}"
+      #{serialize_segments(track)}
+      #{if track.finished?, do: "#EXT-X-ENDLIST", else: ""}
+      """
+  end
+
+  defp serialize_segments(track) do
+    supports_ll_hls? = Track.supports_partial_segments?(track)
+
+    [track.segments, Enum.count(track.segments)..1]
+    |> Enum.zip()
+    |> Enum.flat_map(&do_serialize_segment(&1, supports_ll_hls?))
+    |> Enum.join("\n")
   end
 
-  defp serialize_segment(segment) do
+  defp do_serialize_segment({%Segment{} = segment, idx}, supports_ll_hls?) do
+    [
+      # serialize partial segments just for the last 2 live segments, otherwise just keep the regular segments
+      if(supports_ll_hls? and idx <= 4,
+        do: serialize_partial_segments(segment, idx == 1),
+        else: []
+      ),
+      serialize_regular_segment(segment)
+    ]
+    |> List.flatten()
+  end
```

Comment:
- use Enum.splt_at to extract segments that should be kept partial and the rest
- use a module attribute to keep the number at which we split